src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -145,4 +148,48 @@ public static ResourceTypeSchema load(final JSONObject schemaJson) {
     public Map<String, Object> getUnprocessedProperties() {
         return Collections.unmodifiableMap(this.unprocessedProperties);
     }
+
+    public boolean hasWriteOnlyProperties(final JSONObject resourceModel) {
+        return this.getWriteOnlyPropertiesAsStrings().stream()
+            .anyMatch(writeOnlyProperty -> hasProperty(RegExUtils.removeFirst(writeOnlyProperty, "/properties"), resourceModel));
+    }
+
+    public boolean hasProperty(final String property, final JSONObject resourceModel) {
+        return hasProperty(new JSONPointer(property), resourceModel);
+    }
+
+    public boolean hasProperty(final JSONPointer property, final JSONObject resourceModel) {
+        try {
+            return property.queryFrom(resourceModel) != null;
+        } catch (JSONPointerException e) {
+            return false;
+        }
+    }
+
+    public void removeWriteOnlyProperties(final JSONObject resourceModel) {
+        this.getWriteOnlyPropertiesAsStrings().stream()
+            .forEach(writeOnlyProp -> removeProperty(new JSONPointer(RegExUtils.removeFirst(writeOnlyProp, "/properties")),
+                resourceModel));
+    }
+
+    public static void removeProperty(final JSONPointer property, final JSONObject resourceModel) {
+        String refParts = property.toString();
+        // split pointer into tokens, removing first delimiter
+        List<String> refTokens = Arrays.asList(RegExUtils.removeFirst(refParts, "/").split("/"));
+        final String key = refTokens.get(refTokens.size() - 1);
```

Comment:
think it'd be easier to make the `getRefTokens` method public?

Reply:
what are you referring to here?

Reply:
JSONPointer has a [getRefTokens](https://github.com/everit-org/json-schema/blob/2a445667317f61b28604fb7a3955fba35be49693/core/src/main/java/org/everit/json/schema/JSONPointer.java#L288-L290) method that we could use instead of the hacky approach. would have to change the visibility though, can show you how to do that.